application/src/main/java/uk/nhs/hee/tis/revalidation/service/RecommendationServiceImpl.java

```diff
@@ -254,13 +257,30 @@ public boolean submitRecommendation(String recommendationId, String gmcNumber,
    * @return The last submitted Recommendation or empty
    */
   public TraineeRecommendationRecordDto getLatestRecommendation(String gmcId) {
+    final DoctorsForDB doctorForDB;
+    final var optionalDoctorsForDB = doctorsForDBRepository.findById(gmcId);
+
+    if (optionalDoctorsForDB.isPresent()) {
+      doctorForDB = optionalDoctorsForDB.get();
+    } else {
+      throw new RecommendationException(format(DOCTOR_NOT_FOUND_MESSAGE, gmcId));
+    }
```

Comment:
Rewritting this as 
```
if (optionalDoctorsForDB.isEmpty()) {
  throw new RecommendationException(format(DOCTOR_NOT_FOUND_MESSAGE, gmcId));
}
```
would allow you do a "cleaner" declaration of `final DoctorsForDB doctorForDB = optionalDoctorsForDB.get();` after.